source/client/benchmark_client_impl.cc

```diff
@@ -69,7 +71,9 @@ StatisticPtrMap BenchmarkClientHttpImpl::statistics() const {
 
 bool BenchmarkClientHttpImpl::tryStartRequest(CompletionCallback caller_completion_callback) {
   const uint64_t max_in_flight =
-      max_pending_requests_ + (use_h2_ ? max_active_requests_ : connection_limit_);
+      provide_resource_backpressure_
+          ? max_pending_requests_ + (use_h2_ ? max_active_requests_ : connection_limit_)
+          : UINT64_MAX;
```

Comment:
What about:

```
if (provide_resource_backpressure_) {
  const uint64_t max_in_flight =
      max_pending_requests_ + (use_h2_ ? max_active_requests_ : connection_limit_);

  if (requests_initiated_ - requests_completed_ >= max_in_flight) {
    // When we allow client-side queueing, we want to have a sense of time spend waiting on that
    // queue. So we return false here to indicate we couldn't initiate a new request.
    return false;
  }
}
```

Reply:
Much nicer indeed -- I'm sold :-) Done in 0fcba50e4bbd0a645251ec81a9e377562aa433e6